Add openssl to agent image

mgianatagh opened the pull request:
Add openssl to rancher-agent image.

Successful test run: https://build.verrazzano.io/job/verrazzano/job/pb-vz804/4/

Also manually tested and saw rancher and rancher-agent pods starts successfully.

pfmackin commented:
How did you know to add this new opensll param?  There must be some doc that explains this.

mgianatagh commented:
> How did you know to add this new opensll param? There must be some doc that explains this.

No documentation that I am aware of.  I found out by trying to test with the image in the verrazzano environment, and the pod did not go to ready state.  The pod log showed was trying to execute an openssl command and it was not found.  I added openssl, retested in verrazzano environment and the pod came ready.

pfmackin commented:
I am just wondering what changed where we now need this.  There should be some doc somewhere that tells us to use that param, even though it works it seems risky to use it without having something that tells us why we need it.

mgianatagh commented:
> I am just wondering what changed where we now need this. There should be some doc somewhere that tells us to use that param, even though it works it seems risky to use it without having something that tells us why we need it.

What changed is a couple of weeks ago we switched to using a rancher image that we built from source.  And then no one noticed that the rancher-agent that gets deployed by rancher wasn't working.  So this never worked after we switched to building from source.  So openssl is just one of those items you discover the product needs but is not present in the oraclelinux image we use as base.  Once this fix is done and I update the install, the rancher-agent image will be working for the first time with the one we build from source.

I am also working on an acceptance test change right now to check for these pods so that this condition will get caught in the future.
